model/Service/Lti/LaunchData/EncryptedLtiLaunchData.php

```diff
@@ -0,0 +1,249 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2018 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoEncryption\Service\Lti\LaunchData;
+
+use oat\taoEncryption\Service\Algorithm\AlgorithmSymmetricService;
+use oat\taoEncryption\Service\KeyProvider\SimpleKeyProviderService;
+use oat\taoLti\models\classes\LtiLaunchData;
+use Zend\ServiceManager\ServiceLocatorAwareInterface;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+
+class EncryptedLtiLaunchData extends LtiLaunchData implements ServiceLocatorAwareInterface
```

Comment:
I need this class because only some variables of the LtiLaunchData are encrypted. If I will encrypt everything before creation of LtiLaunchData then the user name will not be displayed properly because the Session User EncryptedLtiUser it's created based on this object.